python/python/bystro/rare_variant/tests/test_parent_of_origin.py

```diff
@@ -0,0 +1,115 @@
+"""
+This code generates synthetic genotype and phenotype data for a genetic study.
+It simulates individuals with or without a mother and/or a father, generates
+genotypes based on minor allele frequency, and creates phenotype data based on
+the genotype and parental status. The function returns a dictionary containing
+various aspects of the generated data.
+"""
+import numpy as np
+import numpy.linalg as la
+from bystro.rare_variant.parent_of_origin import POESingleSNP
+
+
+def cosine_similarity(vec1, vec2):
+    v1 = np.squeeze(vec1)
+    v2 = np.squeeze(vec2)
+    num = np.dot(v1, v2)
+    denom = la.norm(v1) * la.norm(v2)
+    return num / denom
+
+
+def generate_data(beta_m, beta_p, rng, n_individuals=10000, cov=None, maf=0.25):
```

Comment:
You might want to include a quick test for generate_data as well to make sure you're getting what you expect here because another test is dependent on it.

Reply:
For some context, I have a vcf simulator that I use for multiple tests so I have it under test just to make sure if it ever incurs an accidental typo or something that it's still working as expected in any of the tests that rely on it, so just a small test with a small n that tests that you're getting the expected shape/type of data I think would be sufficient.